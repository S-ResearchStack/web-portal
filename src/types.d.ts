declare module '@transcend-io/conflux';
declare module 'native-file-system-adapter';

interface ShowSaveFilePickerOptions {
  suggestedName: string;
}

interface Window {
  showSaveFilePicker: (options?: ShowSaveFilePickerOptions) => Promise<FileSystemFileHandle>;
}

declare module '*.svg' {
  import * as React from 'react';

  export const ReactComponent: React.FunctionComponent<
    React.SVGProps<SVGSVGElement> & { title?: string }
  >;

  export default ReactComponent;
}

interface CSSStyleDeclaration {
  webkitOverflowScrolling: string;
}

declare module '*.svg?url' {
  const src: string;
  export default src;
}

declare namespace NodeJS {
  interface ProcessEnv {
    NODE_ENV: 'production' | 'development' | 'test' | 'none';
    API_URL: string;
    SUPERSET_URL: string;
    // TODO: need to move to aws parameter store
    SUPERSET_ID: string;
    SUPERSET_PASSWORD: string;
    GOOGLE_CLIENT_ID: string;
    AUTHENTICATION_MODE: string;
    GOOGLE_OAUTH_URL: string;
    PUBLIC_PATH: string | undefined;
    MOCK_API: string;
    VERSION: string;
  }
}

declare module 'smoothscroll-polyfill' {
  class Polyfill {
    static polyfill(): void;
  }
  export default Polyfill;
}

declare module 'css-spring';

declare module "*.jpg";
declare module "*.png";
