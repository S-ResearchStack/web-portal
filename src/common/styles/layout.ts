import styled from "styled-components";

export const RowContainer = styled.div<{ flex?: number, width?: string, height?: string }>`
  display: flex;
  flex-direction: row;
  flex: ${p => p.flex ?? 1};
  width: ${p => p.width};
  height: ${p => p.height};
  flex-shrink: 0;
`

export const ColumnContainer = styled.div<{ width?: string, height?: string }>`
  display: flex;
  flex-direction: column;
  width: ${p => p.width};
  height: ${p => p.height};
`

export const PaddedColumnContainer = styled(ColumnContainer)`
  padding-left: 8px;
  padding-right: 8px;
`

export const SizedColumnContainer = styled(PaddedColumnContainer)`
  width: 100%;
`

export const SizedColumnScrollContainer = styled(SizedColumnContainer)`
  width: 100%;
  overflow: hidden;
  overflow-y: auto;
`

export const SizedBox = styled.div<{width?: number, height?: number}>`
  width: ${p => p.width ?? 0};
  height: ${p => p.height ?? 0};
  min-width: ${p => p.width ?? 0};
  min-height: ${p => p.height ?? 0};
  max-width: ${p => p.width ?? 0};
  max-height: ${p => p.height ?? 0};
`
