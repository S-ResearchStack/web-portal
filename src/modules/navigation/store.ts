import { createSelector } from '@reduxjs/toolkit';
import { matchPath } from 'react-router-dom';
import { createBrowserHistory, createMemoryHistory, Location } from 'history';

import { RootState } from 'src/modules/store/store';
import { isInsideTest } from 'src/common/utils/testing';

export enum Path {
  Root = '/',

  Registration = '/registration',
  SignIn = '/signin',
  AccountCreate = '/account/create',

  CreateStudy = '/create-study',

  OverviewSubject = '/overview/participants/:subjectNumber',
  StudyManagement = '/study-management',
  StudyManagementSubject = '/study-management/participants/:subjectNumber',
  StudyManagementEditSurvey = '/study-management/surveys/:surveyId/edit',
  StudyManagementSurveyResults = '/study-management/surveys/:surveyId/results',
  StudyManagementActivityResults = '/study-management/activities/:activityId/results',

  Overview = '/overview',
  SubjectManagement = '/subject-management',
  StudyData = '/study-data',
  StudySettings = '/study-settings',
  LabVisitManagement = '/lab-visit',

  Dashboard = '/dashboard',
  CreateChart = '/dashboard/:dashboardId/chart/create',
  EditChart = '/dashboard/:dashboardId/chart/:chartId/edit',

  TaskManagement = '/task-management',
  CreateSurvey = '/task-management/survey/create',
  CreateActivity = '/task-management/activity/:activityType/create',
  EditActivity = '/task-management/activity/:activityId/edit',

  EducationalManagement = '/educational-management',
  CreateEducational = '/educational-management/:educationType/create',
  EditEducational = '/educational-management/:educationId/edit',
}

export const makeHistory = () =>
  !isInsideTest
    ? createBrowserHistory({ basename: process.env.PUBLIC_PATH })
    : createMemoryHistory();

export const history = makeHistory();

const sectionPaths = [
  Path.Overview,
  Path.Dashboard,
  Path.TaskManagement,
  Path.SubjectManagement,
  Path.StudyData,
  Path.EducationalManagement,
  Path.LabVisitManagement,
  Path.StudySettings
] as const;

export const sectionPathSelector = createSelector(
  (state: RootState) => state.router,
  (r) => sectionPaths.find((s) => matchPath((r.location as unknown as Location).pathname, s))
);
