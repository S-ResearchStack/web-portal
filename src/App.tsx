import React from 'react';
import { Provider } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import { DndProvider } from 'react-dnd';
import { TouchBackend } from 'react-dnd-touch-backend';

import { ThemeProvider } from 'styled-components';

import { store } from 'src/modules/store/store';
import { history, Path } from 'src/modules/navigation/store';
import UserProtectedRoute from 'src/modules/navigation/private-route/UserProtectedRoute';
import MainLayout from 'src/modules/main-layout/MainLayout';
import SignIn from 'src/modules/auth/signin/SignIn';
import SignUp from 'src/modules/auth/signup/SignUp';
import { TooltipProvider, TooltipsList } from 'src/common/components/Tooltip';
import { theme, GlobalStyles } from 'src/styles';

import 'src/modules/auth/authProvider';
import { ModalProvider } from 'src/common/components/Modal';
import buildProvidersTree from './common/components/ProviderTree/ProviderTree';
import RegistrationScreen from 'src/modules/auth/registration/RegistrationScreen';
import TokenProtectedRoute from 'src/modules/navigation/private-route/TokenProtectedRoute';
import CreateStudyScreen from 'src/modules/studies/CreateStudyScreen';


const ProvidersTree = buildProvidersTree([
  [ThemeProvider, { theme: theme }],
  [TooltipProvider, {}],
  [ModalProvider, {}],
  [DndProvider, { backend: TouchBackend, options: { enableMouseEvents: true } }],
  [Provider, { store }]
]);

const App = () => (
  <ProvidersTree>
    <GlobalStyles />
    <ConnectedRouter history={history}>
      <Switch>
        <Route path={Path.SignIn} component={SignIn} />
        <Route path={Path.AccountCreate} component={SignUp} />
        <TokenProtectedRoute path={Path.Registration}>
          <RegistrationScreen />
        </TokenProtectedRoute>
        <UserProtectedRoute path={Path.CreateStudy}>
            <CreateStudyScreen />
        </UserProtectedRoute>
        <UserProtectedRoute>
          <MainLayout />
        </UserProtectedRoute>
      </Switch>
    </ConnectedRouter>
    <TooltipsList />
  </ProvidersTree>
);

export default App;
