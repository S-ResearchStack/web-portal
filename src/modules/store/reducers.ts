import { combineReducers } from '@reduxjs/toolkit';
import { connectRouter } from 'connected-react-router';
import type { Reducer } from 'redux';

import authReducer from 'src/modules/auth/auth.slice';
import studySettingsReducer from 'src/modules/study-settings/studySettings.slice';
import snackbarReducer from 'src/modules/snackbar/snackbar.slice';
import dashboardReducers from 'src/modules/dashboard/dashboard.slice';
import overviewReducers from 'src/modules/overview/overview.slice';
import studiesReducer from 'src/modules/studies/studies.slice';
import studyManagementReducer from 'src/modules/subject/studyManagement.slice';
import educationManagementReducer from 'src/modules/education-management/educationManagement.slice';
import taskManagementReducer from 'src/modules/task-management/taskManagement.slice';
import labVisitsReducers from 'src/modules/lab-visit/labVisit.slice';
import sidebarReducer from 'src/modules/main-layout/sidebar/sidebar.slice';
import fileUploadReducer from 'src/modules/file-upload/fileUpload.slice';
import { history } from 'src/modules/navigation/store';
import { DataSliceState } from 'src/modules/store/createDataSlice';
import studyDataReducer from 'src/modules/study-data/studyData.slice';
import studyRequirementReducer from "src/modules/studies/ParticipationRequirement.slice"
import participantTimezonesReducers from '../common/participantTimezones.slice';

const createReducer = <T, V>(asyncReducers?: { [key: string]: Reducer<DataSliceState<T, V>> }) =>
  combineReducers({
    ...authReducer,
    router: connectRouter(history),
    snackbar: snackbarReducer,
    sidebar: sidebarReducer,
    ...studySettingsReducer,
    ...dashboardReducers,
    ...overviewReducers,
    ...studyManagementReducer,
    ...educationManagementReducer,
    ...taskManagementReducer,
    ...labVisitsReducers,
    ...fileUploadReducer,
    ...asyncReducers,
    ...studiesReducer,
    ...studyDataReducer,
    ...studyRequirementReducer,
    ...participantTimezonesReducers
  });

export default createReducer;
