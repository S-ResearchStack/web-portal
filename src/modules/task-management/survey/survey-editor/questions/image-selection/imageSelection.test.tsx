import React from 'react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components/';
import { TouchBackend } from 'react-dnd-touch-backend';
import { DndProvider } from 'react-dnd';
import { ConnectedRouter } from 'connected-react-router';
import { theme } from 'src/styles';
import { makeStore } from 'src/modules/store/store';
import { makeHistory } from 'src/modules/navigation/store';
import handler from './index';

/* eslint-disable @typescript-eslint/no-explicit-any */

describe('image-selection', () => {
  const expectId = expect.any(String);

  const emptyQuestion = {
    type: 'images',
    id: expectId,
    title: '',
    description: '',
    answers: [
      {
        id: expectId,
        image: '',
        touched: true,
        value: '',
      },
      {
        id: expectId,
        image: '',
        touched: true,
        value: '',
      },
      {
        id: expectId,
        image: '',
        touched: false,
        value: '',
      },
    ],
    options: {
      optional: false,
      imageLabels: false,
      multiSelect: false,
    },
  };

  const notEmptyQuestion = {
    id: 'id',
    type: 'images',
    title: 'Title',
    description: 'Description',
    answers: [
      { id: 'survey_question2', value: 'Label 1', image: 'https://picsum.photos/200', touched: true },
      { id: 'survey_question3', value: 'Label 2', image: 'https://picsum.photos/200', touched: true },
      { id: 'survey_question4', value: '', image: '', touched: false },
    ],
    options: {
      optional: false,
      imageLabels: true,
      multiSelect: false,
    },
  };

  const taskItemFromApi = {
    type: 'CHOICE',
    tag: 'IMAGE',
    id: 'id',
    title: 'Question Title',
    explanation: 'Question Description',
    required: true,
    itemProperties: {
      options: [
        { label: 'Label 1', value: '200' },
        { label: 'Label 2', value: '200' },
      ],
    },
  };

  it('type should be images', async () => {
    expect(handler.type).toEqual('images');
  });

  it('[NEGATIVE] should return empty question', async () => {
    expect(handler.createEmpty()).toEqual(emptyQuestion);
  });

  it('isEmpty should return true', async () => {
    expect(handler.isEmpty(emptyQuestion as any)).toBeTrue();
  });

  it('[NEGATIVE] isEmpty should return true if question is undefined', async () => {
    expect(handler.isEmpty(undefined as any)).toBeTrue();
  });

  it('[NEGATIVE] isEmpty should return true if question answers is undefined', async () => {
    expect(handler.isEmpty({ ...emptyQuestion, answers: undefined } as any)).toBeTrue();
  });

  it('isEmpty should return false', async () => {
    expect(handler.isEmpty(notEmptyQuestion as any)).toBeFalse();
  });

  it('convertFromOtherType should return converted question', async () => {
    const qi = {
      type: 'single',
      id: 'Question1',
      title: 'Title',
      description: 'Description',
      options: {
        optional: false,
        includeOther: false,
      },
      answers: [],
    };
    expect(handler.convertFromOtherType(qi as any)).toEqual({
      ...qi,
      type: 'images',
      options: {
        optional: false,
        imageLabels: false,
        multiSelect: false,
      },
      answers: [
        {
          id: expectId,
          image: '',
          touched: true,
          value: '',
        },
        {
          id: expectId,
          image: '',
          touched: true,
          value: '',
        },
        {
          id: expectId,
          image: '',
          touched: false,
          value: '',
        },
      ],
    });
  });

  it('[NEGATIVE] convertFromOtherType should return empty question', async () => {
    expect(handler.convertFromOtherType({} as any)).toEqual({
      type: 'images',
      id: undefined,
      title: undefined,
      description: undefined,
      answers: [
        {
          id: expectId,
          image: '',
          touched: true,
          value: '',
        },
        {
          id: expectId,
          image: '',
          touched: true,
          value: '',
        },
        {
          id: expectId,
          image: '',
          touched: false,
          value: '',
        },
      ],
      options: {
        optional: false,
        imageLabels: false,
        multiSelect: false,
      },
    });
  });

  it('fromApi should return converted question', async () => {
    expect(handler.fromApi(taskItemFromApi as any)).toEqual({
      type: 'images',
      id: expectId,
      title: 'Question Title',
      description: 'Question Description',
      answers: [
        {
          id: expectId,
          value: 'Label 1',
          touched: true,
          image: '200',
        },
        {
          id: expectId,
          value: 'Label 2',
          touched: true,
          image: '200',
        },
        {
          id: expectId,
          value: '',
          image: '',
          touched: false,
        },
      ],
      options: {
        optional: false,
        imageLabels: true,
        multiSelect: false,
      },
    });
  });

  it('[NEGATIVE] fromApi should return undefined if question type is not an image', async () => {
    expect(
      handler.fromApi({
        ...taskItemFromApi,
        type: '',
        tag: '',
      } as any)
    ).toBeUndefined();
  });

  it('toApi should return converted question', async () => {
    expect(handler.toApi(notEmptyQuestion as any)).toEqual({
      id: expectId,
      type: 'CHOICE',
      tag: 'IMAGE',
      title: 'Title',
      explanation: 'Description',
      required: true,
      itemProperties: {
        options: [{ label: 'Label 1', value: '200' }, { label: 'Label 2', value: '200' },],
      },
    });
  });

  it('[NEGATIVE] toApi should not crash with empty data', async () => {
    expect(handler.toApi({} as any)).toEqual({
      id: undefined,
      type: 'CHOICE',
      tag: 'IMAGE',
      title: undefined,
      explanation: undefined,
      required: false,
      itemProperties: {
        options: [],
      },
    });
  });

  it('renderEditorContent should render editor content', async () => {
    const onChange = jest.fn();
    const confirmOptionRemoval = jest.fn();

    const history: ReturnType<typeof makeHistory> = makeHistory();
    const store: ReturnType<typeof makeStore> = makeStore(history);

    const { baseElement } = render(
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <DndProvider backend={TouchBackend} options={{ enableMouseEvents: true }}>
              {handler.renderEditorContent({
                surveyId: '',
                question: notEmptyQuestion as any,
                onChange,
                confirmOptionRemoval,
              })}
            </DndProvider>
          </ConnectedRouter>
        </Provider>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('[NEGATIVE] renderEditorContent should return null if question is undefined', async () => {
    const onChange = jest.fn();
    const confirmOptionRemoval = jest.fn();

    expect(
      handler.renderEditorContent({
        surveyId: '',
        question: undefined as any,
        onChange,
        confirmOptionRemoval,
      })
    ).toBeNull();
  });
});
