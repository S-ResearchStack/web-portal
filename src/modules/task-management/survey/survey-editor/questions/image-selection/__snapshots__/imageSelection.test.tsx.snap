// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image-selection renderEditorContent should render editor content 1`] = `
.c17 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  background-color: #E3EAFB;
  pointer-events: none;
}

.c15 {
  position: relative;
  z-index: 0;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  -webkit-transition-duration: 0;
  transition-duration: 0;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 1.5rem;
  width: 1.5rem;
  border: none;
  background-color: transparent;
}

.c15:hover {
  cursor: pointer;
}

.c15 > div {
  color: #4475E3;
}

.c15 > div svg {
  fill: #4475E3;
}

.c15:hover:enabled > div {
  color: #2F65E0;
}

.c15:hover:enabled > div svg {
  fill: #2F65E0;
}

.c15:hover:enabled:active > div {
  color: #2057D5;
}

.c15:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c15:disabled > div {
  color: #B3C6F1;
}

.c15:disabled > div svg {
  fill: #B3C6F1;
}

.c15:focus-visible {
  outline: 0.125rem solid #C7D6F7;
}

.c15:focus-visible > div {
  color: #4475E3;
}

.c15:focus-visible > div svg {
  fill: #4475E3;
}

.c15:hover:enabled {
  background-color: #ECF1FC;
}

.c15:hover:enabled:active > div {
  color: #2057D5;
}

.c15:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c15:focus-visible:enabled {
  background-color: #C7D6F7;
  outline: none;
}

.c16 {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translate3d(0,0,0.0625rem);
  -ms-transform: translate3d(0,0,0.0625rem);
  transform: translate3d(0,0,0.0625rem);
}

.c10 {
  -webkit-scrollbar-width: thin;
  -moz-scrollbar-width: thin;
  -ms-scrollbar-width: thin;
  scrollbar-width: thin;
  -webkit-scrollbar-color: #4475E3 #FFFFFF;
  -moz-scrollbar-color: #4475E3 #FFFFFF;
  -ms-scrollbar-color: #4475E3 #FFFFFF;
  scrollbar-color: #4475E3 #FFFFFF;
}

.c10::-webkit-scrollbar {
  width: 1.25rem;
  height: 1.25rem;
}

.c10::-webkit-scrollbar-track {
  background: #FFFFFF;
  border-radius: 0.25rem;
  margin: 0.0625rem 0;
}

.c10::-webkit-scrollbar-thumb {
  background-color: #4475E3;
  border-radius: 0.625rem;
  border: 0.5rem solid #FFFFFF;
  width: 0.25rem;
  min-height: 1.25rem;
}

.c11 {
  resize: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-sizing: border-box;
  outline: none;
  width: 100%;
  padding: 0;
  margin: 0;
  border: none;
  background: #FFFFFF;
  overflow-y: auto;
  caret-color: #4475E3;
  overflow-x: hidden;
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
  min-height: 1.5rem;
}

.c0 {
  display: grid;
  grid-template-columns: repeat(2,11.375rem);
  gap: 1.5rem;
  position: relative;
}

.c5 {
  position: absolute;
  width: 100%;
  height: 100%;
  border: 0.0625rem solid #4475E3;
  border-radius: 0.25rem;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c5:before {
  content: '';
  width: 100%;
  height: 100%;
  position: absolute;
  background-color: #ECF1FC;
  opacity: 0.8;
}

.c7 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  cursor: pointer;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  margin: 0;
  background-image: url(https://picsum.photos/200);
  background-size: cover;
  background-position: center;
  width: 9.375rem;
  height: 9rem;
  border-radius: 0.25rem;
  overflow: hidden;
  position: relative;
  background-color: #F7F8FA;
}

.c3 .c4 {
  opacity: 0;
  -webkit-transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
}

.c3 .c6 svg {
  opacity: 0;
  fill: #4475E3;
  position: relative;
  z-index: 1;
  pointer-events: none;
}

.c3:hover .c4 {
  opacity: 1;
}

.c3:hover .c6 svg {
  opacity: 1;
  fill: #4475E3;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  margin: 0;
  background-image: none;
  background-size: cover;
  background-position: center;
  width: 9.375rem;
  height: 9rem;
  border-radius: 0.25rem;
  overflow: hidden;
  position: relative;
  background-color: rgba(219,219,219,0.2);
}

.c19 .c4 {
  opacity: 0;
  -webkit-transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
}

.c19 .c6 svg {
  opacity: 1;
  fill: #A1A1A1;
  position: relative;
  z-index: 1;
  pointer-events: none;
}

.c19:hover .c4 {
  opacity: 1;
}

.c19:hover .c6 svg {
  opacity: 1;
  fill: #4475E3;
}

.c9 {
  display: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.25rem;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c12 {
  height: 2.625rem;
  overflow: hidden;
  width: 9.375rem;
  color: #474747;
}

.c12::-webkit-input-placeholder {
  color: #474747;
}

.c12::-moz-placeholder {
  color: #474747;
}

.c12:-ms-input-placeholder {
  color: #474747;
}

.c12::placeholder {
  color: #474747;
}

.c20 {
  height: 2.625rem;
  overflow: hidden;
  width: 9.375rem;
  color: #474747;
}

.c20::-webkit-input-placeholder {
  color: #474747;
}

.c20::-moz-placeholder {
  color: #474747;
}

.c20:-ms-input-placeholder {
  color: #474747;
}

.c20::placeholder {
  color: #474747;
}

.c20::-webkit-input-placeholder {
  color: #A1A1A1;
}

.c20::-moz-placeholder {
  color: #A1A1A1;
}

.c20:-ms-input-placeholder {
  color: #A1A1A1;
}

.c20::placeholder {
  color: #A1A1A1;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  width: 11.375rem;
  opacity: 1;
}

.c1 .c13 {
  opacity: 0;
}

.c1:hover .c13 {
  opacity: 1;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  width: 11.375rem;
  opacity: 1;
}

.c18 .c13 {
  opacity: 0;
}

.c18:hover .c13 {
  opacity: 0;
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <label
            class="c3"
            data-testid="label-image"
            height="144"
            src="https://picsum.photos/200"
            width="150"
          >
            <div
              class="c4 c5"
              data-testid="backdrop"
            />
            <div
              class="c6 c7 c8"
            >
              <svg />
            </div>
            <input
              accept="image/*"
              class="c9"
              data-testid="file-input"
              type="file"
            />
          </label>
          <textarea
            class=" c10 c11 c12"
            maxlength="20"
            placeholder="Add label"
            rows="1"
          >
            Label 1
          </textarea>
        </div>
        <div
          class="c13 c14"
        >
          <button
            aria-label="Drag and Drop Item"
            class="c15"
            fill="text"
            style="overflow: hidden; position: relative;"
          >
            <div
              class="c16"
              data-testid="content"
            >
              <svg />
            </div>
            <div
              class="c17"
              color="primaryLightPressed"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
          <button
            aria-label="Delete Item"
            class="c15"
            fill="text"
            style="overflow: hidden; position: relative;"
          >
            <div
              class="c16"
              data-testid="content"
            >
              <svg />
            </div>
            <div
              class="c17"
              color="primaryLightPressed"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
        </div>
      </div>
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <label
            class="c3"
            data-testid="label-image"
            height="144"
            src="https://picsum.photos/200"
            width="150"
          >
            <div
              class="c4 c5"
              data-testid="backdrop"
            />
            <div
              class="c6 c7 c8"
            >
              <svg />
            </div>
            <input
              accept="image/*"
              class="c9"
              data-testid="file-input"
              type="file"
            />
          </label>
          <textarea
            class=" c10 c11 c12"
            maxlength="20"
            placeholder="Add label"
            rows="1"
          >
            Label 2
          </textarea>
        </div>
        <div
          class="c13 c14"
        >
          <button
            aria-label="Drag and Drop Item"
            class="c15"
            fill="text"
            style="overflow: hidden; position: relative;"
          >
            <div
              class="c16"
              data-testid="content"
            >
              <svg />
            </div>
            <div
              class="c17"
              color="primaryLightPressed"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
          <button
            aria-label="Delete Item"
            class="c15"
            fill="text"
            style="overflow: hidden; position: relative;"
          >
            <div
              class="c16"
              data-testid="content"
            >
              <svg />
            </div>
            <div
              class="c17"
              color="primaryLightPressed"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
        </div>
      </div>
      <div
        class="c18"
      >
        <div
          class="c2"
        >
          <label
            class="c19"
            data-testid="label-image"
            height="144"
            src=""
            width="150"
          >
            <div
              class="c4 c5"
              data-testid="backdrop"
            />
            <div
              class="c6 c7 c8"
            >
              <svg />
            </div>
            <input
              accept="image/*"
              class="c9"
              data-testid="file-input"
              type="file"
            />
          </label>
          <textarea
            class=" c10 c11 c20"
            maxlength="20"
            placeholder="Add label"
            rows="1"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
